rust/user-info-fetcher/src/backend/keycloak.rs

```diff
@@ -0,0 +1,190 @@
+use std::collections::HashMap;
+
+use hyper::StatusCode;
+use serde::Deserialize;
+use snafu::{OptionExt, ResultExt, Snafu};
+use stackable_opa_crd::user_info_fetcher as crd;
+use stackable_operator::commons::authentication::oidc;
+
+use crate::{http_error, util::send_json_request, Credentials, UserInfo, UserInfoRequest};
+
+#[derive(Snafu, Debug)]
+pub enum Error {
+    #[snafu(display("unable to log in (expired credentials?)"))]
+    LogIn { source: reqwest::Error },
+
+    #[snafu(display("unable to search for user"))]
+    SearchForUser { source: reqwest::Error },
+
+    #[snafu(display("user with id {user_id:?} was not found"))]
```

Comment:
Nit: We might want to come up with a rule on how we display "dynamic" data in an error message. I always feel like the "dynamic" data should be enclosed in some kind of quotes, for example: `user with id '123' was not found`.

Also, we should maybe aim for a consistent style of error messages like:

- "Unable to find ..." or
- "Failed to find ..."

Reply:
Mixing "unable", "failed", and a completely different form of sentence in one type of error seems a little weird.

Reply:
I agree with quoting variable data.
And yeah, I think "Failed" should be reserved for actual failures (eg: error 500 looking up user, rather than 404, user not found). Maybe we do need some guidelines written.

Reply:
`{:?}` will quote string values. But agreed that it should be written down somewhere.

Reply:
Also agreed with @nicklarsennz about failed.

Reply:
improved error messages in 33e8893